woocommerce/woocommerce
# [e2e tests] fix/playwright-utils-build: Adds build step to `@woocommerce/e2e-playwright-utils`

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes # .

After installing the package, it threw a SyntaxError: Unexpected token 'export' because the add-on setup tries to load the package as a CommonJS module.

I have made the changes such that ESM will be transpiled and will be output to a `/build` directory. I’ve also set the entry point from `src/index.js` to `build/index.js`. This will require a new publish to NPM.

@adimoldovan some files from the root have also been updated when I installed few dependencies in the Utils package.json, so I've committed those as well. Let me know if they shouldn't be added here.

Additionally, please confirm if the package.json is setup correctly to transpile source to build before publishing to NPM, or if any additional changes are required.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. See if the E2E tests pass
2. When publishing to NPM, ensure the `build` directory is present with transpiled files in it.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
